src/module/tests/flows/MatrixTestDataFlow.ts

```diff
@@ -62,5 +79,399 @@ export const MatrixTestDataFlow = {
             if (matrix.hot_sim) pool.addUniquePart('SR5.HotSim', 2);
             if (matrix.running_silent) pool.addUniquePart('SR5.RunningSilent', -2);
         }
-    }
-}
\ No newline at end of file
+    },
+
+    /**
+     * Add Matrix damage to a Test that is a Matrix Attack and will do extra damage based on the number of marks
+     * @param test
+     */
+    addMatrixDamageForTargetMarks(test: SuccessTest) {
+        if (!test.opposed || !test.hasDamage) return;
+        if (!test.hasTestCategory('attack_matrix')) return;
+        const actor = test.actor;
+        if (!actor) return;
+
+        const icon = (test as MatrixTest).icon;
+
+        if (icon) {
+            console.log('processing target', icon);
+            // get the target's persona if the target itself is not an item
+            // TODO n3rf this could probably be cleaned up
```

Comment:
Matrix project todo referencing this line

Reply:
old todo, cleaned up